curvefs/src/client/inode_cache_manager.cpp

```diff
@@ -237,21 +238,36 @@ CURVEFS_ERROR InodeCacheManagerImpl::BatchGetInodeAttr(
 
 CURVEFS_ERROR InodeCacheManagerImpl::BatchGetInodeAttrAsync(
     uint64_t parentId,
-    const std::set<uint64_t> &inodeIds,
+    std::set<uint64_t> *inodeIds,
     std::map<uint64_t, InodeAttr> *attrs) {
-    if (inodeIds.empty()) {
-        return CURVEFS_ERROR::OK;
-    }
-
     NameLockGuard lg(asyncNameLock_, std::to_string(parentId));
     bool ok  = iAttrCache_->Get(parentId, attrs);
     if (ok) {
         return CURVEFS_ERROR::OK;
     }
 
+    // get some inode attr in icache
+    for (auto iter = inodeIds->begin(); iter != inodeIds->end();) {
+        std::shared_ptr<InodeWrapper> inodeWrapper;
+        NameLockGuard lock(nameLock_, std::to_string(*iter));
+        bool ok = iCache_->Get(*iter, &inodeWrapper);
+        if (ok) {
+            InodeAttr tmpAttr;
+            inodeWrapper->GetInodeAttr(&tmpAttr);
+            attrs->emplace(*iter, tmpAttr);
+            iter = inodeIds->erase(iter);
```

Comment:
The return value is ambiguous in different c++ version，so i think It's best not to use it directly